packages/block-editor/src/components/rich-text/embed-handler-picker.native.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { noop } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { forwardRef, useRef, useImperativeHandle } from '@wordpress/element';
+import { Picker } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const DEFAULT_PICKER_OPTIONS = [
+	{
+		id: 'createEmbed',
+		label: __( 'Create embed' ),
+		value: 'createEmbed',
+		onSelect: noop,
+	},
+	{
+		id: 'createLink',
+		label: __( 'Create link' ),
+		value: 'createLink',
+		onSelect: noop,
+	},
+];
+
+export default forwardRef( ( {}, ref ) => {
+	const pickerRef = useRef();
+	const pickerOptions = useRef( DEFAULT_PICKER_OPTIONS ).current;
+
+	function onPickerSelect( value ) {
+		const selectedItem = pickerOptions.find(
+			( item ) => item.value === value
+		);
+		selectedItem.onSelect();
+	}
+
+	useImperativeHandle( ref, () => ( {
+		presentPicker: ( { createEmbed, createLink } ) => {
+			pickerOptions[ 0 ].onSelect = createEmbed;
+			pickerOptions[ 1 ].onSelect = createLink;
+			pickerRef.current?.presentPicker();
+		},
+	} ) );
+
+	return (
+		<Picker
+			ref={ pickerRef }
```

Comment:
the `title` prop is not being set here as the `Picker` is in adherence with the design specs. However, from UI/UX standpoint what I am thinking is the user isn't able to see the URL they have copied until they perform one of the actions ( Create embed or link). Just a thought I wanted to get your feedback on.

Reply:
Good point, I think it's important to display the URL so the user knows what's being pasted. In fact, following the designs, I see that is expected that the URL is displayed as text plain in the Paragraph block once is pasted and we display the picker (see attached screenshot).

<img width="865" alt="Screenshot 2021-09-30 at 11 14 09" src="https://user-images.githubusercontent.com/14905380/135425025-799e8225-41ac-4fee-a2b3-84e5048b6835.png">

Reply:
Oh yes, that is true! I will make the relevant changes so that this UI state is represented. Thanks @fluiddot

Reply:
Resolved in 8f38899